src/main/java/greencity/repository/options/PlaceFilter.java

```diff
@@ -126,11 +126,9 @@ private Predicate hasDiscount(Root<Place> r, CriteriaBuilder cb, FilterDiscountD
             return cb.conjunction();
         }
         return cb.and(
-            cb.equal(r.join("discounts").join("category").get("name"),
-                discount.getCategory().getName()),
-            cb.equal(r.join("discounts").join("specification").get("name"),
+            cb.equal(r.join("discountValues").join("specification").get("name"),
                 discount.getSpecification().getName()),
-            cb.between(r.join("discounts").get("value"),
+            cb.between(r.join("discountValues").get("value"),
```

Comment:
Instead of strings there should be constants. In PlaceFilter many strings with the same name.